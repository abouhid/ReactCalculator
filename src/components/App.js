import React from 'react';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import Calculate from '../logic/calculate';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: null,
      next: null,
      operation: null,
    };
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(btnName) {
    const newState = Calculate(this.state, btnName);
    this.setState(newState);
  }

  render() {
    const { total, next } = this.state;

    return (
      <div className="App">
        <h1>React Calculator</h1>
        <Display amount={next || total || '0'} />
        <ButtonPanel onClick={this.handleClick} />
      </div>
    );
  }
}

export default App;
