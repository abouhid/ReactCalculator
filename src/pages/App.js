import React, { useState } from 'react';
import Display from '../components/Display';
import ButtonPanel from '../components/ButtonPanel';
import Calculate from '../logic/calculate';

const App = () => {
  const [obj, setObj] = useState(
    {
      total: null,
      next: null,
      operation: null,
    },
  );
  const handleClick = btnName => {
    const newState = Calculate(obj, btnName);
    setObj(newState);
  };

  return (

    <div className="App">
      <Display amount={obj.next || obj.total || '0'} />
      <ButtonPanel onClick={handleClick} />
    </div>
  );
};

export default App;
