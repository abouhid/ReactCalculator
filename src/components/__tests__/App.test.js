import React from 'react';
import App from '../../pages/App';
import renderer from 'react-test-renderer';

describe('App', () => {
  it('Tests the rendering of components', () => {
    const tree = renderer
      .create(<App />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
