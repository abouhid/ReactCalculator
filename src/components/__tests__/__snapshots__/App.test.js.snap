// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Tests the rendering of components 1`] = `
<div
  className="App"
>
  <div
    className="display"
  >
    0
  </div>
  <ul
    className="panel"
  >
    <div
      className="panel-button"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
    </div>
    <div
      className="panel-button"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
    </div>
    <div
      className="panel-button"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        %
      </button>
    </div>
    <div
      className="panel-button"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="panel-button"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        7
      </button>
    </div>
    <div
      className="panel-button"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        8
      </button>
    </div>
    <div
      className="panel-button"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        9
      </button>
    </div>
    <div
      className="panel-button"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        X
      </button>
    </div>
    <div
      className="panel-button"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        4
      </button>
    </div>
    <div
      className="panel-button"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        5
      </button>
    </div>
    <div
      className="panel-button"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        6
      </button>
    </div>
    <div
      className="panel-button"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="panel-button"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        1
      </button>
    </div>
    <div
      className="panel-button"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        2
      </button>
    </div>
    <div
      className="panel-button"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        3
      </button>
    </div>
    <div
      className="panel-button"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="panel-button"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        0
      </button>
    </div>
    <div
      className="panel-button"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        .
      </button>
    </div>
    <div
      className="panel-button"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </ul>
</div>
`;
